### PR TITLE
refactor kubelet client to internal folder

### PR DESCRIPTION
**Pull request by pxaws**
**Description:** 
Refactor the logic about kubelet client out of `kubeletstatsreceiver` and put it into the internal folder so that other components can reuse it. For example, the upcoming container insight receiver also call kubelet endpoint to obtain the metadata information about pods: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/3426#discussion_r643520673 

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/2307

**Testing:** 
Unit tests

### CODE REVIEW

**Comment by pxaws** [comment]
@anuraaga @Aneurysm9 @mxiamxia Could you please take a look? Thanks!

**Comment by pxaws** [comment]
@alolita Here is another PR related to AWS Container Insights

**Review by anuraaga — APPROVED** [review]

**Review by mxiamxia — APPROVED** [review]
Thanks!

File: internal/kubelet/go.mod [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module github.com/open-telemetry/opentelemetry-collector-contrib/internal/kubelet
+
+go 1.15
```
Comment (mx-psi) [inline internal/kubelet/go.mod:3]:
```suggestion
go 1.16
```

File: internal/kubelet/go.mod [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+require (
+	github.com/open-telemetry/opentelemetry-collector-contrib/internal/k8sconfig v0.0.0-00010101000000-000000000000
+	github.com/stretchr/testify v1.7.0
+	go.opentelemetry.io/collector v0.27.0
```
Comment (mx-psi) [inline internal/kubelet/go.mod:8]: This should be
```suggestion
	go.opentelemetry.io/collector v0.27.1-0.20210602134904-017aa1dad750
```

File: internal/kubelet/go.mod (hunk 1 above)
Comment (pxaws) [reply internal/kubelet/go.mod]: Already made the change.

File: internal/kubelet/go.mod (hunk 2 above)
Comment (pxaws) [reply internal/kubelet/go.mod]: Already made the change.

**Review by mx-psi — APPROVED** [review]

**Pull request merged**